### ISSUE TITLE
tokio refuses all TCP connections besides the first one

### ISSUE THREAD

**Issue by pyfisch** [tokio-rs/tokio#26]
## Steps to reproduce

1. Compile test case and run it with `cargo run`
2. Run `telnet localhost 5000` to connect to the server. Connection is established and closed immediately.
3. Run `telnet localhost 5000` again. Connection is refused. 

## Expected behavior
Tokio accepts multiple connections, even at the same time.

## Test case

```rust
extern crate tokio;

use std::io;

use tokio::server::listen;
use tokio::reactor::{Reactor, Task, Tick};

struct Connection;

impl Task for Connection {
    fn tick(&mut self) -> io::Result<Tick> {
        Ok(Tick::Final)
    }
}

fn main() {
    let reactor = Reactor::default().unwrap();
    let address = "0.0.0.0:5000".parse().unwrap();
    listen(&reactor.handle(), address, |_| Ok(Connection)).unwrap();
    reactor.run().unwrap();
}
```

**Comment by daschl** [comment]
I couldn't reproduce the issue on tokio hash 618f6f6ac9a107c5f9e65b87b591a2f9bba61a5f (osx)

**Comment by carllerche** [comment]
Fixed by 17f390cd86268a0dcd59b4b9893ae2dcd383ea68

**Comment by pyfisch** [comment]
:+1: Thanks for the fast fix!
